src/node_errors.cc

```diff
@@ -451,7 +451,7 @@ TryCatchScope::~TryCatchScope() {
     if (message.IsEmpty())
       message = Exception::CreateMessage(env_->isolate(), exception);
     ReportFatalException(
```

Comment:
I think we should take `CanContinue()` into account here (that is, don't enhance if `CanContinue()` is false).

cc @nodejs/v8 can you confirm it's safe to call into JavaScript in the destructor of a TryCatch subclass?

Reply:
i had some code a while ago that called into js in the TryCatchScope destructor and it worked fine.

Reply:
Should be fine. The parent `TryCatch` would not be destructed yet, so it would just catch exceptions.